version.go

```diff
@@ -32,16 +33,32 @@ var (
 // Besides returning the Debian upstream version, the "upstream" struct
 // struct fields u.version, u.commitIsh, u.hasRelease and u.isRelease
 // are also set.
+// `preferredRev` should be empty if there are no user preferences.
 // TODO: also support other VCS
-func pkgVersionFromGit(gitdir string, u *upstream, forcePrerelease bool) (string, error) {
+func pkgVersionFromGit(gitdir string, u *upstream, preferredRev string, forcePrerelease bool) (string, error) {
 	var latestTag string
 	var commitsAhead int
 
+	var cmd *exec.Cmd
+
+	// If the user specifies a valid tag as the preferred revision, that tag should be used without additional heuristics.
+	if u.rr != nil {
+		if out, err := u.rr.VCS.Tags(gitdir); err == nil && slices.Contains(out, preferredRev) {
+			latestTag = preferredRev
+			goto FoundLatestTag
+		}
+	}
+
```

Comment:
These are some comments regarding why I use a `goto` statement. Please let me know if there are better alternatives.

I want to use the `git describe` heuristics as a fallback in case I cannot verify if the user specified git revision is valid or not.

The logic I want to implement is:
```
if (u.rr is valid (which makes u.rr.VCS.Tags meaningful) and the user specified tag exists as a valid tag) {
    set the latest tag to this user specified tag
} else {
    the fallback heuristics
}
```

However, to implement the "if" part of the above pseudocode, I cannot find a way to implement it as a single if clause. This will result in the "else" part being written out twice, and I want to avoid such duplication.

In other words, I am trying to deal with the problem in this post: https://stackoverflow.com/questions/13197425/multiple-if-with-one-else

William